servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/DefaultMultiAddressUrlHttpClientBuilderTest.java

```diff
@@ -138,75 +137,98 @@ public HttpHeaders newHeaders() {
+                     .initializer((scheme, address, builder) ->
+                             builder.executionStrategy(internalCtx.executionStrategy())
+                                     .executor(internalCtx.executor())
+                                     .ioExecutor(internalCtx.ioExecutor())
+                                     .bufferAllocator(internalCtx.bufferAllocator())
+                                     .appendClientFilter(client -> {
+                                         actualInternalCtx.set(client.executionContext());
+                                         return new StreamingHttpClientFilter(client) {
+                                         };
+                                     }))
+                     .executor(externalCtx.executor())
+                     .ioExecutor(externalCtx.ioExecutor())
+                     .executionStrategy(externalCtx.executionStrategy())
+                     .bufferAllocator(externalCtx.bufferAllocator())
+                     .buildBlocking()) {
 
             // Check external client
-            final HttpExecutionContext executionContext = streamingHttpClient.executionContext();
-            assertThat(executionContext.executor(), equalTo(CTX.executor()));
-            assertThat(executionContext.executionStrategy(), equalTo(externalExecutionStrategy));
-            assertThat(executionContext.ioExecutor(), equalTo(CTX.ioExecutor()));
-            assertThat(executionContext.bufferAllocator(), equalTo(CTX.bufferAllocator()));
-
+            assertExecutionContext(externalCtx, blockingHttpClient.executionContext());
             // Make a request to trigger the filter execution that extracts the execution context.
-            final HostAndPort address = HostAndPort.of((InetSocketAddress) serverContext.listenAddress());
-            streamingHttpClient.reserveConnection(streamingHttpClient.get("http://" + address)).toFuture().get();
-
+            HttpResponse response = blockingHttpClient.request(
+                    blockingHttpClient.get("http://" + serverHostAndPort(serverContext)));
+            assertThat(response.status(), is(OK));
             // Check internal client
-            assertThat(actualInternalBufferAllocator.get(), equalTo(internalBufferAllocator));
-            assertThat(actualInternalExecutor.get(), equalTo(INTERNAL_CLIENT_CTX.executor()));
-            assertThat(actualInternalIoExecutor.get(), equalTo(INTERNAL_CLIENT_CTX.ioExecutor()));
-            assertThat(actualInternalExecutionStrategy.get(), equalTo(internalExecutionStrategy));
-
-            streamingHttpClient.closeAsync().toFuture().get();
+            assertExecutionContext(internalCtx, actualInternalCtx.get());
         }
     }
 
+    private static void assertExecutionContext(HttpExecutionContext expected, HttpExecutionContext actual) {
+        assertThat(actual, is(notNullValue()));
+        assertThat(actual.ioExecutor(), equalTo(expected.ioExecutor()));
```

Comment:
I wonder if these should be slightly stronger `sameInstance` rather than `equalTo`